Makefile

```diff
@@ -17,154 +17,163 @@
+export PATH := $(UT_CORE_DIR)/toolchain:$(PATH)
 
-# defaults for target arm
-ifeq ($(TARGET),arm)
-CUNIT_VARIANT=arm-rdk-linux-gnueabi
-#CC := arm-rdk-linux-gnueabi-gcc -mthumb -mfpu=vfp -mcpu=cortex-a9 -mfloat-abi=soft -mabi=aapcs-linux -mno-thumb-interwork -ffixed-r8 -fomit-frame-pointer
-# CFLAGS will be overriden by Caller as required
-INC_DIRS += $(UT_CORE_DIR)/sysroot/usr/include
-TARGET = arm
-else
-TARGET = linux
-endif
+# Set target based on TARGET variable
+TARGET := $(if $(TARGET),$(TARGET),linux)
 export TARGET
 
-$(info TARGET [$(TARGET)])
-
 # Moveable Directories based on the caller makefile
 TOP_DIR ?= $(UT_CORE_DIR)
 BIN_DIR ?= $(TOP_DIR)/build/bin
-LIB_DIR ?= $(TOP_DIR)/build/${TARGET}/lib
+LIB_DIR ?= $(TOP_DIR)/build/$(TARGET)/lib
 BUILD_DIR ?= $(TOP_DIR)/build/$(TARGET)/obj
 
 # Non-Moveable Directories
 FRAMEWORK_DIR = $(UT_CORE_DIR)/framework
 UT_CONTROL = $(FRAMEWORK_DIR)/ut-control
 
+# Flags
 XCFLAGS := $(KCFLAGS)
+INC_DIRS += $(UT_CORE_DIR)/include $(UT_CONTROL)/include $(UT_CORE_DIR)/src
+TARGET_EXEC ?= hal_test
 
-# Enable CUnit Requirements
-XCFLAGS += -DUT_CUNIT
-CUNIT_DIR +=  $(FRAMEWORK_DIR)/CUnit-2.1-3/CUnit
-CUNIT_SRC_DIRS += $(CUNIT_DIR)/Sources
-INC_DIRS += $(CUNIT_DIR)/Headers
-SRC_DIRS += $(CUNIT_SRC_DIRS)/Framework
-
-INC_DIRS += $(UT_CORE_DIR)/include
-INC_DIRS += $(UT_CONTROL)/include
-INC_DIRS += $(UT_CORE_DIR)/src
-
-SRC_DIRS += $(UT_CORE_DIR)/src
-
-XLDFLAGS += -L $(UT_CONTROL)/build/$(TARGET)/lib -lut_control
-
-MKDIR_P ?= @mkdir -p
-
-# Defaults for target linux
-ifeq ($(TARGET),linux)
-CUNIT_VARIANT=i686-pc-linux-gnu
-CC := gcc -ggdb -o0 -Wall
+# Set up compiler and source directories based on VARIANT
+ifneq ($(VARIANT),CPP) # CUNIT case
+  VARIANT = C
+
+  # Enable CUnit Requirements
+  XCFLAGS += -DUT_CUNIT
+  CUNIT_DIR = $(FRAMEWORK_DIR)/CUnit-2.1-3/CUnit
+  INC_DIRS += $(CUNIT_DIR)/Headers $(UT_CORE_DIR)/src/c_source
+  SRC_DIRS += $(CUNIT_DIR)/Sources/Framework $(UT_CORE_DIR)/src
+  XLDFLAGS += $(YLDFLAGS) $(LDFLAGS) -L$(UT_CONTROL)/build/$(TARGET)/lib -lut_control -Wl,-rpath, -pthread -lpthread
+
+  # Source files
+  SRCS := $(shell find $(SRC_DIRS) -name *.c -or -name *.s)
+else # GTEST case
+  VARIANT = CPP
+
+  SRC_DIRS += $(UT_CORE_DIR)/src
+  EXCLUDE_DIRS = $(SRCDIR)/c_source
+  GTEST_SRC = $(FRAMEWORK_DIR)/gtest/$(TARGET)/googletest-1.15.2
+  INC_DIRS += $(GTEST_SRC)/googletest/include $(UT_CORE_DIR)/src/cpp_source $(UT_CORE_DIR)/src
+  GTEST_LIB_DIR = $(UT_CORE_DIR)/build/$(TARGET)/gtest/lib/
+  XLDFLAGS += $(YLDFLAGS) $(LDFLAGS) -L$(UT_CONTROL)/build/$(TARGET)/lib -L$(GTEST_LIB_DIR) -lgtest_main -lgtest -lut_control -lpthread
+
+  # Source files
+  SRCS := $(shell find $(SRC_DIRS) -type f \( -name '*.cpp' -o -name '*.c' \) | grep -v "$(EXCLUDE_DIRS)")
 endif
 
-XLDFLAGS += -Wl,-rpath, $(YLDFLAGS) $(LDFLAGS) -pthread  -lpthread
-
-SRCS := $(shell find $(SRC_DIRS) -name *.cpp -or -name *.c -or -name *.s)
+ifeq ($(TARGET),arm)
+CUNIT_VARIANT = arm-rdk-linux-gnueabi
```

Comment:
Compilers are not related to `CUNIT` you've called the compiler `CUNIT_VARIANT`

so you can remove this now, because it's not used?